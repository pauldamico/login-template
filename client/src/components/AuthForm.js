import React, { useState } from "react";


export default function AuthForm(props) {
  const {clearError, error, loginToggle, authenticateUser, loginToggler } = props;
  const [userInfo, setUserInfo] = useState({ username: "", password: "" });

  function formChangeHandler(event) {
    const { value, name } = event.target;
    setUserInfo((prev) => ({ ...prev, [name]: value }));
  }
  function formLogin(event) {
    clearError()
    event.preventDefault();
    authenticateUser(userInfo)    
  }

  return (
    <div className="auth-div">
      <form onSubmit={formLogin} className="auth-form">
        <label>Username </label>
        <input data-testid="username-input" name="username" onChange={formChangeHandler} type="text" />
        <label>Password </label>
        <input name="password" onChange={formChangeHandler} type="text" />
         <button >{loginToggle ? "Login" : "Sign Up"}</button>
        <button type="button" onClick={()=>{loginToggler()}}>Back</button>        
      </form>
      {error}
    </div>
  );
}
