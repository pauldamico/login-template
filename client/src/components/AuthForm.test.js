import React from 'react'
import renderer from 'react-test-renderer';
import {render} from "@testing-library/react"
import "@testing-library/jest-dom"
import AuthForm from './AuthForm'

describe("SnapShot of AuthForm", ()=>{
    it('renders correctly', () => {
        const tree = renderer
          .create(<AuthForm/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
})


describe("username input ", ()=>{
it("Username Input Exist Test", ()=>{
const {getByTestId} = render(<AuthForm />)
const input = getByTestId("username-input")
expect(input).toBeTruthy()
}    )

it("empty string", ()=>{
    const {getByTestId} = render(<AuthForm />)
    const input = getByTestId("username-input")
    expect(input.textContent).toBe('')
    }    )

})