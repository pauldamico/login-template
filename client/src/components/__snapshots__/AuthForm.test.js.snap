// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot of AuthForm renders correctly 1`] = `
<div
  className="auth-div"
>
  <form
    className="auth-form"
    onSubmit={[Function]}
  >
    <label>
      Username 
    </label>
    <input
      data-testid="username-input"
      name="username"
      onChange={[Function]}
      type="text"
    />
    <label>
      Password 
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="text"
    />
    <button>
      Sign Up
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
  </form>
</div>
`;
